app/uk/gov/hmrc/cataloguefrontend/metrics/connector/MetricsConnector.scala

```diff
@@ -0,0 +1,140 @@
+import cats.implicits.none
+import com.google.inject.ImplementedBy
+import play.api.Logger
+import play.api.libs.json.Reads
+import uk.gov.hmrc.cataloguefrontend.metrics.model._
+import uk.gov.hmrc.http.HttpClient
+import uk.gov.hmrc.play.bootstrap.config.ServicesConfig
+
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+@ImplementedBy(classOf[MetricsConnector.Impl])
+trait MetricsConnector {
+  def query(
+             maybeGroup: Option[GroupName],
+             maybeName: Option[DependencyName],
+             maybeRepository: Option[RepositoryName]
+           ): Future[MetricsResponse]
+
+  // will return only the ones that have repos that have at least one dependency
+  def getAllGroups: Future[Seq[Group]]
+
+  // will return only the ones that have at least one dependency
+  def getAllRepositories: Future[Seq[Repository]]
+
+  def getAllDependencies: Future[Seq[DependencyName]]
+}
+
+object MetricsConnector{
+
+  abstract class ViaQuery(implicit ec: ExecutionContext) extends MetricsConnector {
+    override def getAllGroups: Future[Seq[Group]] = query(none, none, none)
+      .map(_.metrics)
+      .map(Group.apply)
+
+    override def getAllRepositories: Future[Seq[Repository]] = query(none, none, none)
+      .map(_.metrics)
+      .map(Repository.apply)
+
+    override def getAllDependencies: Future[Seq[DependencyName]] = query(none, none, none)
+      .map(_.metrics)
+      .map(DependencyName.apply)
+  }
+
+  class Impl @Inject() (
+    httpClient: HttpClient,
+    servicesConfig: ServicesConfig
+  )(implicit val ec: ExecutionContext) extends ViaQuery {
+    import uk.gov.hmrc.http._
+
+    private implicit val hc: HeaderCarrier = HeaderCarrier()
+    private val platformProgressMetricsBaseURL: String = servicesConfig.baseUrl("platform-progress-metrics")
+    implicit val rF: Reads[MetricsResponse] = MetricsResponse.reads
+    val logger                           = Logger(this.getClass)
+
+    import cats.syntax.option._
+
+    override def query(maybeGroup: Option[GroupName], maybeName: Option[DependencyName], maybeRepository: Option[RepositoryName]): Future[MetricsResponse] = {
+      val url = url"$platformProgressMetricsBaseURL/platform-progress-metrics/metrics" + Monoid.combineAll(
+        List(
+          maybeGroup.map(g => s"group=${g.value}&"),
+          maybeName.map(n => s"name=${n.value}&"),
+          maybeRepository.map(r => s"repository=${r.value}")
+        )
+      )
+        .map(params => s"?$params")
+        .orEmpty
+
+      httpClient
+        .GET[MetricsResponse](
+          url
+        )
+        .recoverWith {
+          case UpstreamErrorResponse.Upstream5xxResponse(x) =>
+            logger.error(s"An error occurred when connecting to serviceDependencies. baseUrl: $platformProgressMetricsBaseURL", x)
+            Future.successful(MetricsResponse(Seq.empty))
+        }
+    }
+  }
+
+  class Mocked (implicit val ec: ExecutionContext) extends ViaQuery {
```

Comment:
Do we still need the Mock stuff? It can be moved out to the `catalogue-data-stub` repo if it is.

Reply:
done.